.buildkite/pipeline-pull-request.yml

```diff
@@ -1,213 +1,307 @@
+# `yq 'explode(.)' .buildkite/pipeline-pull-request.yml` to see expanded anchor/aliases
+container:
+  kubernetes: &kubernetes
+    sidecars:
+    - image: us-west1-docker.pkg.dev/ci-compute/buildkite-images/buildkite-dind:v1
+      volumeMounts:
+        - mountPath: /var/run/
+          name: docker-sock
+      securityContext:
+        privileged: true
+        allowPrivilegeEscalation: true
+    mirrorVolumeMounts: true # CRITICAL: this must be at the same indentation level as sidecars
+    podSpec: &podSpec
+      containers:
+        - &commandContainer
+          image: us-west1-docker.pkg.dev/ci-compute/buildkite-images/buildkite-command-container:v2
+          command:
+          - |-
+            echo "Command step was not overridden."
+            exit 1
+          volumeMounts:
+            - mountPath: /var/run/
+              name: docker-sock
+          resources:
+            requests:
+              cpu: 7500m
+              memory: 30G
+      volumes:
+      - name: docker-sock
+        emptyDir: {}
+
+agents:
+  queue: buildkite-gcp
+
 steps:
   - label: "fossa analyze"
-    agents:
-      queue: "init"
-      docker: "*"
     command: "./scripts/buildkite/fossa.sh"
+
   - label: ":golang: unit test"
-    agents:
-      queue: "workers"
-      docker: "*"
-    commands:
-      - "make .just-build" # ensure that we are not rebuilding binaries and not regenerating code
-      - "CASSANDRA_HOST=cassandra make install-schema && make cover_profile" # make install-schema is needed for a server startup test. See main_test.go
-      - "./scripts/buildkite/gen_coverage_metadata.sh .build/coverage/metadata.txt"
     artifact_paths:
       - ".build/coverage/*.out"
       - ".build/coverage/metadata.txt"
     retry:
       automatic:
         limit: 1
+    agents:
+      queue: workers
```

Comment:
if this job is failing on K8s for pull-request pipeline it will also probably fail in master pipeline.